.github/workflows/terraform-on-branch-push.yml

```diff
@@ -0,0 +1,79 @@
+name: Terraform tools on Branch Push
+on:
+  push:
+    branches-ignore: 
+      - main
+jobs:
+   docs:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+
+    - name: Render terraform docs inside the README.md and push changes back to PR branch
+      uses: terraform-docs/gh-actions@v1.3.0
+      with:
+        working-dir: .
+        output-file: README.md
+        output-method: inject
+        git-push: "true"
+   terraform-fmt:
+      runs-on: ubuntu-latest
+      permissions:
+         contents: write
+      steps:
+       - name: Checkout repository
+         uses: actions/checkout@v4
+   
+       - name: Setup Terraform
+         uses: hashicorp/setup-terraform@v2
+         with:
+            terraform_version: 1.3.0
```

Comment:
This version of terraform is super old, I think we should just not specify this to always use the latest version.